src/main/drivers/accgyro_fake.c

```diff
@@ -51,54 +34,65 @@
 #include "accgyro_fake.h"
 
 static int16_t fakeGyroADC[XYZ_AXIS_COUNT];
+gyroDev_t *fakeGyroDev;
 
 #if defined(SIMULATOR_BUILD) && defined(SIMULATOR_MULTITHREAD)
-static pthread_mutex_t gyroUpdateLock;
-#endif
-#if defined(SIMULATOR_BUILD) && defined(SIMULATOR_GYRO_SYNC)
-static bool gyroUpdated = false;
+static pthread_mutex_t gyroLock;
```

Comment:
Agreed. You can also define `inline` lock and unlock functions in the header file.